rust/src/main.rs

```diff
@@ -65,6 +67,58 @@ async fn inner_async_main(args: Vec<String>) -> Result<i32> {
     .await?
 }
 
+/// Multicall entrypoint for `ostree-container`.
+async fn multicall_ostree_container(args: Vec<String>) -> Result<i32> {
+    ostree_ext::cli::run_from_iter(args).await?;
+    Ok(0)
+}
+
+/// Dispatch multicall binary to relevant logic, based on callname from `argv[0]`.
+async fn dispatch_multicall(callname: String, args: Vec<String>) -> Result<i32> {
+    match callname.as_str() {
+        // TODO(lucab): fix the final callname for this after ostree side at
+        // https://github.com/ostreedev/ostree/issues/2480 is settled.
+        "ostree-container" => multicall_ostree_container(args).await,
+        "rpm-ostree" | _ => inner_async_main(args).await,
+    }
+}
+
+/// Process a string from `argv[0]` into a clean callname.
+fn callname_from_argv0(argv0: &str) -> &str {
+    // SAFETY: `rsplit` iterator always contains at least one item.
+    let callname = argv0.rsplit('/').into_iter().nth(0).unwrap();
```

Comment:
There is as of relatively recently https://doc.rust-lang.org/std/primitive.str.html#method.rsplit_once

At the time it was too new to use, but that's no longer the case it looks like :tada: 

See e.g. https://github.com/ostreedev/ostree-rs-ext/pull/187

Reply:
Ah nice, I've never used that before but it fits perfectly.